src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/XmlDescriptor.php

```diff
@@ -435,6 +436,12 @@ private function getArgumentNodes(array $arguments, \DOMDocument $dom)
                 foreach ($this->getArgumentNodes($argument->getValues(), $dom) as $childArgumentXML) {
                     $argumentXML->appendChild($childArgumentXML);
                 }
+            } elseif ($argument instanceof ServiceLocatorArgument) {
+                $argumentXML->setAttribute('type', 'service_locator');
```

Comment:
the XML format uses a dash, not a underscore. So we should do the same for the descriptor